recipes/scons/all/conanfile.py

```diff
@@ -0,0 +1,93 @@
+from conans import ConanFile, tools
+from conans.errors import ConanException, ConanInvalidConfiguration
+import io
+import os
+import shutil
+import sys
+
+
+class SConsConan(ConanFile):
+    name = "scons"
+    description = "SCons is an Open Source software construction tool—that is, a next-generation build tool"
+    license = "MIT"
+    url = "https://github.com/conan-io/conan-center-index/"
+    homepage = "https://scons.org"
+    topics = ("conan", "scons", "build", "configuration", "development")
+    settings = "os"  # Added to let the CI test this package on all os'es
+
+    _autotools = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def configure(self):
+        # Detect availability of a python interpreter
+        # FIXME: add a python build requirement
+        if not tools.which("python"):
+            raise ConanInvalidConfiguration("This recipe requires a python interpreter.")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename("scons-{}".format(self.version), self._source_subfolder)
+
+    @property
+    def _python_executable(self):
+        return "python"
```

Comment:
should this be 'python3' ? scons requires at least python 3.5

Reply:
nvm, scons requires python to be named 'python' i think.

Reply:
This is where a python recipe will come in handy.
I have one for 3.7 but it's not working for WIndows right now.
And I should update it for 3.8.

Reply:
Is there any limitation using sys.executable here?

Reply:
No, the current recipe does not use `sys.executable` at all.
It assumes `python` is present.

Maybe it requires a `cpython` (build) requirement?

Reply:
this doesn't align with how existing recipes are using python (such as boost, pybind, meson, etc.). I believe most of users already have python and they expect the same interpreter to be used. I think (it might be completely wrong) usage of .exe installer for conan is very low, and most people install it via pip, so python is already there.

Reply:
I side with @SSE4 .
People who use the windows conan exe installer, should add cpython to their profile as a build requirement.
Currently, the same goes for people not having cmake installed.

Reply:
What I don't know, is whether python is always present.
Now python2 is EOF, python3 is in use.
Do newer linux distribution always have `python` present? Or should it use `python3`?

Reply:
> People who use the windows conan exe installer, should add cpython to their profile as a build requirement.
Currently, the same goes for people not having cmake installed.

I agree.

> Do newer linux distribution always have python present?

Yes, Python3 as Python 2 is EOL. I would prefer call "python" in this case, because is calls the default one. In my case, I have both installed, but I use `alias python=python3`, otherwise, my system uses python2 as default.

Reply:
To add another data point, the meson recipe also implicitly requires a python interpreter.

Reply:
this is totally okay, there are plenty of recipes (include meson) which rely on python. it's definitely correct assumption, as conan itself requires python, and Linux/OSX usually have python out of the box anyway. 
we have similar assumptions about other tools - e.g. we usually believe that shell (bash), make, perl, autotools are available on *nix systems. we also assume cmake is available everywhere.
so for me it's not a big deal, otherwise we might need to change almost everything to the paranoid mode. let's be practical here.